packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -148,8 +181,23 @@ export function createComponentExtension<
 export function createReactExtension<
   T extends (props: any) => JSX.Element | null,
 >(options: {
+  /**
+   * A loader or synchronously supplied component that is rendered by this extension.
+   */
   component: ComponentLoader<T>;
+
+  /**
+   * Additional component data that is attached to the top-level extension component.
```

Comment:
Maybe put in a `@link` to the extraction function of that data?

Reply:
Eh, cba right now, this is pretty internal anyway x) could add that though ye